programs/clearing_house/src/controller/orders.rs

```diff
@@ -851,17 +860,24 @@ fn fulfill_order(
     }
 
     let mut base_asset_amount = 0_u128;
-    let mut potentially_risk_increasing = false;
+    let mut quote_asset_amount = 0_u128;
     let mut order_records: Vec<OrderRecord> = vec![];
     for fulfillment_method in fulfillment_methods.iter() {
-        if user.orders[user_order_index].status != OrderStatus::Open {
+        if user.orders[user_order_index].status != OrderStatus::Open
+            || quote_asset_amount >= max_quote_asset_amount
+        {
             break;
         }
 
+        let max_base_asset_amount = max_base_asset_amount
```

Comment:
so as explainer:
`max_base_asset_amount` is how much the user could fill at max leverage and goes down after another fulfillment has occurred

should probably try to be smarter in `determine_fulfillment_methods` based on price